dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileHandler.java

```diff
@@ -0,0 +1,146 @@
+    }
+  }
+
+  @Override
+  public void startElement(
+      String namespaceUri, String localName, String qualifiedName, Attributes attributes)
+      throws SAXException {
+    if (!namespaceUri.isEmpty()) {
+      throw new SAXException(
+          "unrecognized element: " + qualifiedName + " in namespace " + namespaceUri);
+    }
+    switch (localName) {
+      case "LinkageCheckerFilter":
+        break;
+      case "LinkageError":
+        stack.push(new LinkageErrorMatcher());
+        break;
+      case "Source":
+        SourceMatcher sourceMatcher = new SourceMatcher();
+        ((LinkageErrorMatcher) stack.peek()).setSourceMatcher(sourceMatcher);
+        stack.push(sourceMatcher);
+        break;
+      case "Target":
+        TargetMatcher targetMatcher = new TargetMatcher();
+        ((LinkageErrorMatcher) stack.peek()).setTargetMatcher(targetMatcher);
+        stack.push(targetMatcher);
+        break;
+      case "Package":
+        PackageMatcher packageMatcher = new PackageMatcher(attributes.getValue("name"));
+        addMatcherToParent(packageMatcher);
+        break;
+      case "Class":
+        String classNameOnClass = attributes.getValue("name");
+        ClassMatcher classMatcher = new ClassMatcher(classNameOnClass);
+        addMatcherToParent(classMatcher);
+        break;
+      case "Method":
+        String classNameOnMethod = attributes.getValue("className");
+        MethodMatcher methodMatcher =
+            new MethodMatcher(classNameOnMethod, attributes.getValue("name"));
+        addMatcherToParent(methodMatcher);
+        break;
+      case "Field":
+        String classNameOnField = attributes.getValue("className");
+        FieldMatcher fieldMatcher = new FieldMatcher(classNameOnField, attributes.getValue("name"));
+        addMatcherToParent(fieldMatcher);
+        break;
+      default:
+        throw new SAXException("Unknown tag " + localName);
+    }
+  }
+
+  @Override
+  public void endElement(String namespaceUri, String localName, String qualifiedName)
+      throws SAXException {
+    SymbolProblemMatcher poppedMatcher;
+    if (!namespaceUri.isEmpty()) {
+      throw new SAXException(
+          "unrecognized element: " + qualifiedName + " in namespace " + namespaceUri);
+    }
+    switch (localName) {
+      case "Source":
+      case "Target":
+        poppedMatcher = stack.pop();
+        if (!(poppedMatcher instanceof TargetMatcher)
+            && !(poppedMatcher instanceof SourceMatcher)) {
+          throw new SAXException("Unexpected matcher in stack");
+        }
+        break;
+      case "LinkageError":
+        poppedMatcher = stack.pop();
+        if (!(poppedMatcher instanceof LinkageErrorMatcher)) {
+          throw new SAXException("Unexpected stack status after reading LinkageError element");
+        }
+        matchers.add((LinkageErrorMatcher) poppedMatcher);
+    }
+  }
+
+  @Override
+  public void endDocument() throws SAXException {
+    if (stack.size() != 0) {
```

Comment:
`!stack.isEmpty()`

Reply:
Good catch. But endDocument is no longer needed.